src/bun.js/api/server.zig

```diff
@@ -2138,7 +2138,7 @@ fn NewRequestContext(comptime ssl_enabled: bool, comptime debug_mode: bool, comp
         pub fn renderMetadata(this: *RequestContext) void {
             var response: *JSC.WebCore.Response = this.response_ptr.?;
             var status = response.statusCode();
-            var needs_content_range = this.needs_content_range;
+            var needs_content_range = this.needs_content_range and this.sendfile.remain < this.blob.size();
```

Comment:
`sendfile` is initialized to `undefined`, so we need to be extremely careful when it is accessed

I think this is okay because `needs_content_range` is initialized to `false`

Reply:
The next line does:
```zig
const size = if (needs_content_range)
    this.sendfile.remain
else
    this.blob.size();
```
Hence I won't be introducing any new NPE here :smirk: